src/hotspot/share/code/codeCache.cpp

```diff
@@ -1147,8 +1308,8 @@ void CodeCache::old_nmethods_do(MetadataClosure* f) {
     length = old_compiled_method_table->length();
     for (int i = 0; i < length; i++) {
       CompiledMethod* cm = old_compiled_method_table->at(i);
-      // Only walk alive nmethods, the dead ones will get removed by the sweeper or GC.
-      if (cm->is_alive() && !cm->is_unloading()) {
+      // Only walk !is_unloading nmethods, the other ones will get removed by the GC.
+      if (!cm->is_unloading()) {
```

Comment:
I thought we were going to walk the unloading methods?  Or should that be another bug fix?  It's unlikely to ever find a reproducer so I wouldn't mind if it was fixed with this giant patch.

Reply:
You are correct. However, I think the bug fix should be in a separate bug fix, and not in this RFE.